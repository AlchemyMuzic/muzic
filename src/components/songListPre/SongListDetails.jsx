import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

export default function SongListDetails({ songs }) {

  const songElements = songs.map(song => (
    <li key={song.title}>
      <Link to={`/songs/${song.id}`}>
        <h2>{song.title}</h2>
      </Link>
    </li>
  ));

  return (
    <ul data-testid="songTest">
      {songElements}
    </ul>
  );
}

SongListDetails.propTypes = {
  songs: PropTypes.arrayOf(PropTypes.shape({
    title: PropTypes.string.isRequired,
  }))
};
