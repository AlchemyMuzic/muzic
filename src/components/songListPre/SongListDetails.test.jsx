import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { render, cleanup } from '@testing-library/react';
import SongListDetails from './SongListDetails';

describe('SongListDetails component', () => {
    afterEach(() => cleanup());
    it('renders a list of artists', () => {
        const { asFragment } = render(<BrowserRouter><SongListDetails
            songs={[]} /></BrowserRouter>);
        expect(asFragment()).toMatchSnapshot();
    })
})
