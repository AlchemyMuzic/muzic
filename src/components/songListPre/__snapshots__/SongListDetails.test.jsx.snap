// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SongListDetails component renders a list of artists 1`] = `
<DocumentFragment>
  <ul
    data-testid="songTest"
  />
</DocumentFragment>
`;
